recipes/ncbi-cxx-toolkit-public/26.0/conanfile.py

```diff
@@ -0,0 +1,1587 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanException
+from conans.errors import ConanInvalidConfiguration
+import os
+import shutil
+import subprocess
+import tempfile
+import sys
+import stat
+
+def _remove_readonly(fn, path, excinfo):
+    try:
+        os.chmod(path, stat.S_IWRITE)
+        fn(path)
+    except Exception as exc:
+        print("Skipped: ", path, " because ", exc)
+
+class NcbiCxxToolkit(ConanFile):
+    name = "ncbi-cxx-toolkit-public"
+    license = "CC0-1.0"
+    homepage = "https://ncbi.github.io/cxx-toolkit"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "NCBI C++ Toolkit"
+    topics = ("ncbi", "biotechnology", "bioinformatics", "genbank", "gene",
+              "genome", "genetic", "sequence", "alignment", "blast",
+              "biological", "toolkit", "c++")
+    settings = "os", "compiler", "build_type", "arch"
+
+    tk_tmp_tree = ""
+    tk_src_tree = ""
+    all_NCBI_requires = {}
+
+    options = {
+        "shared":     [True, False],
+        "sharedDeps": [True, False],
+        "fPIC":       [True, False],
+        "projects":   "ANY",
+        "targets":    "ANY",
+        "tags":       "ANY"
```

Comment:
These don't look like actual options, are they?

Reply:
There are few (unusual?) design decisions. You see, the Toolkit is huge. It builds about 230 libraries (even much more at NCBI). Usually, a user does not need them all. That is why, we use filters. At NCBI, it is often "projects", in Conan, I fancy, it is mostly "targets". "Target" is a library which should be built. The toolkit is smart enough to find all dependencies and build them too.
This also creates a problem with dependencies. One set of libraries, may require "libdb", another - not. That is why the Toolkit recipe finds dependencies dynamically. First, it runs the configuration is special mode and gets the list of dependencies. Then, it maps "Toolkit" requirements to what is available. After Conan installs what it was asked to install, the toolkit builds what it can build. So, for example, if GRPC is not available, targets that require it will not be built.

The problem with the Toolkit is that the entry price is pretty high. One has to download and build a set of third party libraries, configure this, configure that. People do need it outside of NCBI, but it is not easy. The purpose of using Conan (and Conan center) is to reduce this price. Make initial steps (much) easier.